settings/src/components/auto-tabbing-input.js

```diff
@@ -9,27 +9,13 @@ import { useCallback } from '@wordpress/element';
 import NumericControl from './numeric-control';
 
 const AutoTabbingInput = ( props ) => {
-	const { inputs, setInputs, onComplete, error } = props;
+	const { inputs, setInputs, error, setError } = props;
 
 	const handleChange = useCallback( ( value, event, index, inputRef ) => {
 		setInputs( ( prevInputs ) => {
 			const newInputs = [ ...prevInputs ];
 
-			// Clean input
-			if ( value.trim() === '' ) {
-				event.target.value = '';
-				value = '';
-			}
-
-			newInputs[ index ] = value;
-
-			// Check if all inputs are filled
-			const allFilled = newInputs.every( ( input ) => '' !== input );
-			if ( allFilled && onComplete ) {
-				onComplete( true );
-			} else {
-				onComplete( false );
-			}
+			newInputs[ index ] = value.trim() === '' ? '' : value;
```

Comment:
Rather than modifying the value of the DOM element here, we can simply update the `inputs`, and the data will flow down to the DOM.

The `complete` state can also be derived from the `inputs` values, rather than checking on every change to an input, see the check here https://github.com/WordPress/wporg-two-factor/pull/154/files#diff-9b176e953d93ade24d65eef39d9a5ebb31bec9e284b349cf8ff313f6e9869f8fR231